client/src/main/scala/org/http4s/client/oauth1/OAuthConfig.scala

```diff
@@ -0,0 +1,15 @@
+package org.http4s.client.oauth1
+
+import org.http4s.client.oauth1.Header.{Callback, Verifier}
+import org.http4s.client.oauth1.{Header => OAuthHeader}
+
+case class OAuthConfig(
```

Comment:
Are we sure more config won't be necessary?  We've started to move away from case classes for configuration, so we can add more parameters later without breaking binary compatibility.  This is not an issue if this is truly a closed set, but it makes me a bit nervous.

Reply:
my rule of thumb is usually if number of params > 3 i use case classes

Reply:
i never thought about your point

Reply:
It's tedious to write out binary compatible case classes by hand, but it does improve the compatibility story.  If we think this will always be enough config, we can probably leave this alone.

An interesting project has emerged to help with this, but should probably be discussed separately: [data-class](https://github.com/alexarchambault/data-class).